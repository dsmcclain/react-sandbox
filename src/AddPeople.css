.AddPeople {
    margin: 20px;
    padding: 20px;
    border: 2px solid #ff999;
    background: #e46161;
    text-align: center;
}