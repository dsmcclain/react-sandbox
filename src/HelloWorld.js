import React, { Component } from 'react';
import './HelloWorld.css';

class HelloWorld extends Component {
  constructor(props) {
      super(props);
      this.state = { 
        greeting: 'Hello',
        fredCount: 0,
      };
      this.fredify = this.fredify.bind(this);
      this.removePeople = this.removePeople.bind(this);
  }

  fredify() {
      const fredness = this.state.greeting;
      if (fredness === 'Hello') {
        this.setState({ greeting: 'Yaba Daba Doo' });
        this.setState({ fredCount: this.state.fredCount + 1 });
      } else {
        this.setState({ greeting: 'Hello' });
      };
  }

  removePeople() {
      this.props.removePeople(this.props.name);
  }
  
  render() {
    return (
      <div className="HelloWorld">
        {this.state.greeting} {this.props.name}!
        <br />
        You have been Fredified {this.state.fredCount} times.
        <br />
        <button className="fredify" onClick={this.fredify}>Fredify!</button>
        <button className="remove" onClick={this.removePeople}>Remove Me!</button>
      </div>
    );
  }
}

export default HelloWorld;