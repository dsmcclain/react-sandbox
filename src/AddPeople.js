import React, { Component } from 'react';
import './AddPeople.css';

class AddPeople extends Component {
    constructor(props) {
        super(props);
        this.state = { greetingName: '' };
        this.handleUpdate = this.handleUpdate.bind(this);
        this.addPeople = this.addPeople.bind(this);
    }

    addPeople() {
        this.props.addPeople(this.state.greetingName);
        this.setState({ greetingName: '' });
    }

    handleUpdate(e) {
        this.setState({ greetingName: e.target.value });
    }

    render() {
        return (
            <div className="AddPeople">
                <input 
                    type="text" 
                    onChange={this.handleUpdate}
                    value={this.state.greetingName}
                />
                &nbsp;&nbsp;
                <button onClick={this.addPeople}>Add</button>
            </div>
        );
    }
}

export default AddPeople;