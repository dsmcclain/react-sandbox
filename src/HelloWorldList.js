import React, { Component } from 'react';
import './HelloWorldList.css';
import AddPeople from './AddPeople';

import WidgetFrame from './WidgetFrame';
import HelloWorld from './HelloWorld';

class HelloWorldList extends Component {
    constructor(props) {
        super(props);
        this.state = { people: ['Jim', 'Sally', 'Bender'] };
        this.addPeople = this.addPeople.bind(this);
        this.removePeople = this.removePeople.bind(this);
    }

    renderGreetings() {
        return this.state.people.map(name => (
            <WidgetFrame title={name} children = {
                <HelloWorld 
                key={name} 
                name={name}
                removePeople={this.removePeople}
                />
            } />
        ));
    }

    addPeople(newName) {
        this.setState({ people: [...this.state.people, newName] });
    }

    removePeople(removeName) {
        const filteredPeople = this.state.people.filter(name => {
            return name !== removeName;
        });
        this.setState({ people:filteredPeople });
    }

    render() {
        return (
            <div className="HelloWorldList">
                <AddPeople addPeople={this.addPeople} />
                {this.renderGreetings()}
            </div>
        );
    }
}

export default HelloWorldList;