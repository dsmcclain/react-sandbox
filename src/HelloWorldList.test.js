import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import AddPeople from './AddPeople';
import HelloWorld from './HelloWorld';

describe(HelloWorldList, () => {
    const component = shallow(
        <HelloWorldList />
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWorldList />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('contains an AddPeople subcomponent', () => {
        expect(component.find(AddPeople)).toHaveLength(1);
    });

    it('contains the same number of HelloWorld components as people', () => {
        const helloWorlds = component.find(HelloWorld).length;
        const people   = component.state('people').length;
        expect(helloWorlds).toEqual(people);
    });

    it('adds another person when the addPeople function is called', () => {
        const before = component.find(HelloWorld).length;
        component.instance().addPeople('Sample');
        component.update();
        const after = component.find(HelloWorld).length;
        expect(after).toBeGreaterThan(before);
    });

    it('removes a person from the list when the removePerson function is called', () => {
        const before = component.find(HelloWorld).length;
        const removeMe = component.state('people')[0];
        component.instance().removePeople(removeMe);
        const after = component.find(HelloWorld).length;
        expect(after).toBeLessThan(before);
    });
})