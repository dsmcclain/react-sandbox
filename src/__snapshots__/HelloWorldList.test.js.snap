// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldList renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddPeople"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Jim
    !
    <br />
    <button
      className="fredify"
      onClick={[Function]}
    >
      Fredify!
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Sally
    !
    <br />
    <button
      className="fredify"
      onClick={[Function]}
    >
      Fredify!
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Bender
    !
    <br />
    <button
      className="fredify"
      onClick={[Function]}
    >
      Fredify!
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
</div>
`;
